rotkehlchen/db/dbhandler.py

```diff
@@ -2875,60 +2892,36 @@ def query_owned_assets(self) -> List[Asset]:
         # but think on the performance. This is a synchronous api call so if
         # it starts taking too much time the calling logic needs to change
         cursor = self.conn.cursor()
-        query = cursor.execute(
-            'SELECT DISTINCT currency FROM timed_balances ORDER BY time ASC;',
-        )
 
         results = set()
-        for result in query:
+        for table_entry in TABLES_WITH_ASSETS:
```

Comment:
I am not really in love with this approach with `TABLES_WITH_ASSETS`?

Indeed there can be assets in all these tables. The reason I was using `timed_balances` is because that means that the assets were snapshotted. And we take a snapshot every day so if they are in any of those others tables they should also be in the snapshots.

Indeed your way is the safest as there can be edge cases where a snapshot was not taken but I am afraid of eventual slowdown with this.

Let's try to also measure it performance wise on a big DB like mine.